packages/devtools_app/lib/src/memory/memory_heap_tree_view.dart

```diff
@@ -640,7 +649,78 @@ class HeapTreeViewState extends State<HeapTree>
     );
   }
 
-  Widget _buildAllocationsControls(ThemeData themeData) {
+  AnimationController _setupBubbleAnimationController() {
+    // Setup animation controller to handle the update bubble.
+    const animationDuration = Duration(milliseconds: 500);
+    final bubbleAnimation = AnimationController(
+      duration: animationDuration,
+      reverseDuration: animationDuration,
+      upperBound: 15.0,
+      vsync: this,
+    );
+
+    bubbleAnimation.addStatusListener(_animationStatusListener);
+
+    return bubbleAnimation;
+  }
+
+  void _animationStatusListener(AnimationStatus status) {
+    if (status == AnimationStatus.completed) {
+      // Reverse from larger bubble back to normal bubble radius.
+      _animation.reverse();
+    }
+  }
+
+  static const _updateCircleRadius = 8.0;
+
+  Timer removeUpdateBubble;
+
+  Widget displayTimestampUpdateBubble() {
+    final themeData = Theme.of(context);
```

Comment:
nit: load theme data right when it is needed rather from this widget's context.

Reply:
Good idea, don't think it changes anything.